modules/op-creating-pipeline-run-using-pipelines-as-code.adoc

```diff
@@ -240,43 +293,36 @@ git commit --amend --no-edit && git push --force-with-lease
+== Specifying automatic cancellation-in-progress for a pipeline run
 
-You can use the temporary installation token generated by {pac} from GitHub App to access the GitHub API. The token value is stored in the temporary `{{git_auth_secret}}` dynamic variable generated for private repositories in the `git-provider-token` key.
+By default, {pac} does not cancel pipeline runs automatically. Every pipeline run that {pac} creates and starts executes until it completes.
 
-For example, to add a comment to a pull request, you can use the `github-add-comment` task from {tekton-hub} using a {pac} annotation:
+However, events that trigger pipeline runs can come in quick succession. For example, if a pull request triggers a pipeline run and then the user updates the pull request with a new commit, a new copy of the pipeline run might start every time. If several pushes happen, several copies can run, resulting in a drain on cluster resources.
 
-[source,yaml]
-----
-...
-  pipelinesascode.tekton.dev/task: "github-add-comment"
-...
-----
+You can configure a pipeline run to enable automatic cancellation-in-progress. If you enable automatic cancellation for a pipeline run, {pac} cancels the pipeline run in the following situations:
 
-You can then add a task to the `tasks` section or `finally` tasks in the pipeline run definition:
+* {pac} has successfully started a copy of the same pipeline run for the same pull request or the same source branch.
+
+* The pull request that triggered the pipeline run is merged or closed.
+
+You can use the following example to enable automatic cancellation for the `sample-pipeline` pipeline run:
 
 [source,yaml]
 ----
-[...]
-tasks:
-  - name:
-      taskRef:
-        name: github-add-comment
-      params:
-        - name: REQUEST_URL
-          value: "{{ repo_url }}/pull/{{ pull_request_number }}" <1>
-        - name: COMMENT_OR_FILE
-          value: "Pipelines as Code IS GREAT!"
-        - name: GITHUB_TOKEN_SECRET_NAME
-          value: "{{ git_auth_secret }}"
-        - name: GITHUB_TOKEN_SECRET_KEY
-          value: "git-provider-token"
-...
+apiVersion: tekton.dev/v1
+kind: PipelineRun
+metadata:
+  name: sample-pipeline
+annotations:
+  pipelinesascode.tekton.dev/cancel-in-progress: "true"
+# ...
 ----
-<1> By using the dynamic variables, you can reuse this snippet template for any pull request from any repository.
 
 [NOTE]
 ====
-On GitHub Apps, the generated installation token is available for 8 hours and scoped to the repository from where the events originate unless configured differently on the cluster.
+{pac} cancels a pipeline run _after_ starting a new copy of this pipeline run successfully. The `pipelinesascode.tekton.dev/cancel-in-progress` setting does _not_ ensure that only one copy of the pipeline run is executing at any time.
 ====
+
+:FeatureName: Automatic cancellation-in-progress of pipeline runs
+include::snippets/technology-preview.adoc[]
```

Comment:
Should this be at the top of the section?

Reply:
we have several of these in this reference module and they all are at the bottoms of their subsections

Reply:
There's apparently no guidance about snippet placement in the openshift-docs contribution guide, but I'd caution against replicating that pattern. You could run into situations where users follow instructions only to be surprised that a feature is a technology preview.